src/web/app/svgedit/svgcanvas.ts

```diff
@@ -4674,33 +4657,47 @@ export default $.SvgCanvas = function (container: SVGElement, config: ISVGConfig
     if (!tempGroup || !tempGroup.contains(elem)) {
       return;
     }
+
     const originalLayer = getCurrentDrawing().getLayerByName(elem.getAttribute('data-original-layer'));
     const currentLayer = getCurrentDrawing().getCurrentLayer();
     const targetLayer = originalLayer || currentLayer;
+
+    // to explicitly remove one element from the temp group layers
+    selectedLayers = selectedLayers.filter((_layer, index, array) => !(array.indexOf(elem.getAttribute('data-original-layer')) === index));
```

Comment:
In my opinion, putting `array.indexOf` in `filter` may increase the time complexity from O(n) to O(n^2).
How about moving `indexOf` out? e.g.
```
const idx = selectedLayers.indexOf(elem.getAttribute('data-original-layer'));
if (idx >= 0) selectedLayers.splice(idx, 1);
```